### PR TITLE
doc/glossary: add "Bucket"

### PR DESCRIPTION
**Pull request by zdover23**
Add "Bucket" to the glossary. The definition added here is just a first attempt, but a man's reach must exceed his grasp, or what's a heaven for?

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by anthonyeleven — APPROVED** [review]

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/50224 - Quincy backport
https://github.com/ceph/ceph/pull/50225 - Pacific backport

File: doc/glossary.rst [hunk 1]
```diff
@@ -14,6 +14,16 @@
                 system interface. Since Luminous (12.2), BlueStore has been
                 Ceph's default and recommended storage back end.
 
+        Bucket
```
Comment (cbodley) [inline doc/glossary.rst:17]: bucket is a term from aws S3, so it's probably helpful to refer to some of their documentation. for example,

https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-buckets-s3.html
https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingBucket.html

i don't think it's useful to say anything about zones or multisite in the basic definition of a bucket

Comment (zdover23) [reply doc/glossary.rst:17]: I'll remove all of this what-contains-what-relative-to-each-other stuff when I have "object", "bucket", "zone", "zonegroup", and "realm" defined in the glossary. Think of this as lexicographical scaffolding.

Comment (zdover23) [reply doc/glossary.rst:17]: I'll add the AWS references after a nap.

Comment (anthonyeleven) [reply doc/glossary.rst:17]: I think about that sometimes when writing about RGW, but my sense is that so few people are using Swift that it wasn't worth the complexity. ymmv.

Comment (zdover23) [reply doc/glossary.rst:17]: https://github.com/ceph/ceph/pull/50246 - this PR adds AWS and OpenStack information about what RGW calls "buckets" (and this PR adds the information to the definition of bucket that Casey wanted me to add) (and it also removes the zone-and-zonegroup-related where-in-the-hierarchy-do-buckets-fall information that Casey deemed to be irrelevant and which, after a night's sleep and greater familiarity with the family of concepts here discussed, I have come around on)

**Pull request merged**